pkg/controller/registry/resolver/resolver.go

```diff
@@ -114,10 +123,6 @@ func (r *Resolver) Resolve(namespaces []string, subs []*v1alpha1.Subscription) (
 
 	r.addInvariants(namespacedCache, variables)
 
-	if err := namespacedCache.Error(); err != nil {
```

Comment:
Would removing this error check here cause all errors to get overlooked, versus just those related to catalog connectivity? Are we risking some other kind of error case sneaking through and causing issues in the `Solve()` function below? 

Maybe to be safe, we can introduce a custom error type that is specific for catalog connectivity issues, and purposefully drop that error here?